### PR TITLE
cloudflare_zone: add verification_key attribute

### LINKED ISSUE #280: Support for Creating Partial (CNAME) Zones
### Terraform Version
Terraform v0.11.11

### Affected Resource(s)
-cloudflare_zone

## Support creating partial (CNAME) Zones: 
Please add support for creating and managing partial (CNAME) zones. 

https://support.cloudflare.com/hc/en-us/articles/360020615111-Configuring-a-CNAME-setup
https://support.cloudflare.com/hc/en-us/articles/360020348832-Understanding-a-CNAME-Setup

### API Call to Create Partial Zone
curl -X POST "https://api.cloudflare.com/client/v4/zones" \
     -H "X-Auth-Email: user@example.com" \
     -H "X-Auth-Key: API-KEY" \
     -H "Content-Type: application/json" \
     --data '{"type": "partial", "name":"example.com","jump_start":true,"account":{"id":"ACCOUNTID"}}'

### PR DESCRIPTION
**Pull request by inge4pres**
This supersedes #492, fixes #280

### CODE REVIEW

File: cloudflare/resource_cloudflare_zone_test.go [hunk 1]
```diff
@@ -73,6 +73,28 @@ func TestAccCloudflareZonePartialSetup(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareZonePartialSetupEnterprise(t *testing.T) {
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_zone_test.go]: This test currently fails as the `plan` isn't enterprise but free

```
------- Stdout: -------
=== RUN   TestAccCloudflareZonePartialSetupEnterprise
--- FAIL: TestAccCloudflareZonePartialSetupEnterprise (1.62s)
    testing.go:569: Step 0 error: Check failed: Check 3/5 error: cloudflare_zone.tf-acc-partial-setup-zone: Attribute 'plan' expected "enterprise", got "free"
FAIL
```

I think this is related to #488 though

Comment (inge4pres) [reply cloudflare/resource_cloudflare_zone_test.go]: uh ok, will look into that issue
does Travis run the acceptance tests?

Comment (jacobbednarz) [reply cloudflare/resource_cloudflare_zone_test.go]: Afraid not; it's kicked off by TeamCity which has limited maintainer access and doesn't report here 🙁 

You can however, run these from your local machine using some [instructions from the README](https://github.com/terraform-providers/terraform-provider-cloudflare#developing-the-provider)

Comment (inge4pres) [reply cloudflare/resource_cloudflare_zone_test.go]: Ok thanks, I was indeed able to see that fail using our test account... I suspect fixing that involves a change in the core of API?

Comment (jacobbednarz) [reply cloudflare/resource_cloudflare_zone_test.go]: I’ve opened #511 to hopefully fix it. Stay tuned!

**Comment by patryk** [comment]
We're good with this one despite the test failing (for now)?

**Comment by inge4pres** [comment]
Hi @patryk I was able to setup an Enterprise zone with a custom built provider. I'd like to do an additional test to validate if #488 is impacting here too, then your call or @jacobbednarz for when it's safe to merge 😄

**Comment by jacobbednarz** [comment]
I'm on the fence; I'd really rather not have the integration tests failing on master as that is helpful to know whether changes in the codebase or Cloudflare API have introduced regressions.

**Comment by jacobbednarz** [comment]
This now passes locally but not in CI 🤔

**Comment by inge4pres** [comment]
@jacobbednarz what kind of error arises in CI? I was able to do an E2E creation of Enterprise zone using the provider built with this version

**Comment by jacobbednarz** [comment]
The failure is 

```
------- Stdout: -------
=== RUN   TestAccCloudflareZonePartialSetupEnterprise
--- FAIL: TestAccCloudflareZonePartialSetupEnterprise (11.13s)
    testing.go:569: Step 0 error: Check failed: Check 3/5 error: cloudflare_zone.tf-acc-partial-setup-zone: Attribute 'plan' expected "enterprise", got "free"
FAIL
```

But locally, it works as expected. I haven't had a chance to dig into this yet though. Might need another merge from master to get the partial support stuff in here?

**Comment by inge4pres** [comment]
Hi @jacobbednarz I rebased from master including chages that fix #488 can you please verify this goes through CI?
Thanks 😄

**Comment by jacobbednarz** [comment]
Re-applied this in #532 as this one is messy due to merge updates.

**Pull request closed without merging**